core/src/main/java/org/testcontainers/dockerclient/DockerClientConfigUtils.java

```diff
@@ -17,42 +17,43 @@
     // See https://github.com/docker/docker/blob/a9fa38b1edf30b23cae3eade0be48b3d4b1de14b/daemon/initlayer/setup_unix.go#L25
     public static final boolean IN_A_CONTAINER = new File("/.dockerenv").exists();
 
-    @Getter(lazy = true)
```

Comment:
Any reason why you dropped lazily cached result? This change will make it run every time somebody queries the host ip, and I don't see how does it fix it.
Also, was it broken?

Reply:
my bad you are right i will refactor it to lazy cache again.
and yes it is broken if we run it in GitLab CI directly in the build container with the DinD service it resolves to the default gateway, however it should resolve to tcp://docker:2375 or just docker

Reply:
oh wow, nice finding!